hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalConditionEval.java

```diff
@@ -37,6 +40,10 @@
 
     private static final long serialVersionUID = 1L;
 
+    private enum TypeValue {
+        DATA, EVENT
+    }
```

Comment:
I'm not sure. I'm +1 to change to ValueType.
As you suggest, we can use the presence of the String/Event as implicit discriminator about it.
I'm not against that either, just wondering in some corner cases (if any) what happen if there is a null value on both (then perhaps is an IllegalArgumentException).
I'm for maintain temporarily this enum changing to ValueType and we can clean it later once the events alerter is putting in place (we can review it).